### PR TITLE
treehouses rename input check (fixes #315)

### LINKED ISSUE #315: Underscore character disappears in treehouses rename
treehouses rename wifi_name becomes wifiname

### PR DESCRIPTION
**Pull request by irisb1701**
## Description
Fixes error with inputting underscore/special characters when changing hostname

checks for
-special characters
-spaces
-length greater than 64
-no input

So that it conforms closely to debian hostname perferences shown below

## Screenshot
Screen Shot 2019-08-18 at 4 08 12 PM

### CODE REVIEW

File: modules/rename.sh [hunk 1]
```diff
@@ -1,11 +1,20 @@
 #!/bin/bash
 
 function rename () {
-  CURRENT_HOSTNAME=$(< /etc/hostname tr -d " \\t\\n\\r")
-  echo "$1" > /etc/hostname
-  sed -i "s/127.0.1.1.*$CURRENT_HOSTNAME/127.0.1.1\\t$1/g" /etc/hosts
-  hostname "$1"
-  echo "Success: the hostname has been modified"
+  if
+    [ "$(expr "$1" : ".*[!@#\$%^\&*()_+].*")" -gt 0 ] || #checks for special characters
```
Comment (dogi) [inline modules/rename.sh]: @irisb1701 think we need to check if $1 is composed of a-z A-z 0-9 and - cause in ascii there are more then this special characters

Comment (irisb1701) [reply modules/rename.sh]: Ok just made the changes

**Comment by dogi** [comment]
`a-z`, `A-Z` and `-`

**Comment by dogi** [comment]
```
root@treehouses:~/cli# git checkout renamefix 
Branch renamefix set up to track remote branch renamefix from origin.
Switched to a new branch 'renamefix'
root@treehouses:~/cli# ./cli.sh rename dogi-hiro
Unsuccessful: Make sure to remove special characters and spaces.
```

`dogi-hiro` is a valid hostname and shloud be possible

**Comment by dogi** [comment]
@irisb1701 
```
root@treehouses:~/cli# hostname dogi-hiro
root@treehouses:~/cli# logout
Connection to 192.168.0.134 closed.
nemo@omen:~$ ssh root@192.168.0.134                                                                                                                                                                                               
Linux dogi-hiro 4.14.98+ #1200 Tue Feb 12 20:11:02 GMT 2019 armv6l                                                                                                                                                                
                                                                                                                                                                                                                                  
The programs included with the Debian GNU/Linux system are free software;                                                                                                                                                         
the exact distribution terms for each program are described in the                                                                                                                                                                
individual files in /usr/share/doc/*/copyright.                                                                                                                                                                                   
                                                                                                                                                                                                                                  
Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent                                                                                                                                                                 
permitted by applicable law.                                                                                                                                                                                                      
Last login: Mon Aug 26 01:49:25 2019 from 192.168.0.115                                                                                                                                                                           
                                                                                                                                                                                                                                  
SSH is enabled and the default password for the 'pi' user has not been changed.                                                                                                                                                   
This is a security risk - please login as the 'pi' user and type 'passwd' to set a new password.                                                                                                                                  
                                                                                                                                                                                                                                  
root@dogi-hiro:~# 
```

**Comment by dogi** [comment]
from wikipedia
```
Hostnames are composed of a series of labels concatenated with dots. For example, "en.wikipedia.org" is a hostname. Each label must be from 1 to 63 characters long,[2] and the entire hostname including the delimiting dots, but not a trailing dot, has a maximum of 253 ASCII characters.[3]

The Internet standards (Requests for Comments) for protocols mandate that component hostname labels may contain only the ASCII letters 'a' through 'z' (in a case-insensitive manner), the digits '0' through '9' and the hyphen-minus character ('-'). The original specification of hostnames in RFC 952 mandated that labels could not start with a digit or with a hyphen-minus character and could not end with a hyphen-minus. However, a subsequent specification (RFC 1123) permitted hostname labels to start with digits.
```

**Pull request merged**